src/stdlib.c

```diff
@@ -251,6 +251,23 @@ Array.prototype.lastIndexOf = function(a, b) {
   return -1;
 };
 ));
+
+  v7_exec(v7, &res, STRINGIFY(
+    Array.prototype.reduce = function(a, b) {
+      var f = 0; var r = b; var t = this;
```

Comment:
let's just mutate the argument `b`.
Let's also use `this`. Minification doesn't pay off here. 13 bytes for decl to save 9 bytes. `this` is already passed, `t` takes extra space on the environment and AST-code is longer (`AST_THIS` is a special opcode)